mesonbuild/compilers.py

```diff
@@ -1758,8 +1758,8 @@ def get_include_args(self, path, is_system):
         return ['-I' + path]
 
     def get_warn_args(self, level):
-        if level == '2':
-            return ['-wi']
+        if level == '2' or level == '3':
```

Comment:
GDC is covered, DMD's help says:
```
-w             warnings as errors (compilation will halt)
-wi            warnings as messages (compilation will continue)
```
while LDC says:
```
-w                                      - Enable warnings
-wi                                     - Enable informational warnings
```

I think there is a bug in LDC if -w is really -Werror, either in the docs or in its implementation. In any case, it looks like I need to clarify this with upstream and then adjust Meson again. :-/